packages/dev/core/src/Engines/WebGL/webGLPipelineContext.ts

```diff
@@ -110,32 +170,45 @@ export class WebGLPipelineContext implements IPipelineContext {
     }
 
     /**
-     * @param uniformName
-     * @param x
-     * @param y
+     * @param _uniformName
+     * @param _x
+     * @param _y
+     * @param _z
+     * @param _w
      * @hidden
      */
-    public _cacheFloat2(uniformName: string, x: number, y: number): boolean {
-        let cache = this._valueCache[uniformName];
+    public _cacheFloatN(_uniformName: string, _x: number, _y: number, _z?: number, _w?: number): boolean {

```

Comment:
should we make _y optional and use it for setFloat and setInt as well ?

Reply:
There is no cache for setInt and setFloat

Reply:
would be great to add no ?

Reply:
Added :-)